src/magic_folder/snapshot.py

```diff
@@ -210,6 +324,124 @@ def deserialize_dict(snapshot_dict, author):
 
         return deserialize_dict(local_snapshot_dict, author)
 
+@attr.s
+class RemoteSnapshot(object):
+    """
+    Represents a snapshot corresponding to a particular version of a
+    file authored by a particular human.
+
+    :ivar name: the name of this Snapshot. This is a mangled path
+        relative to our local magic-folder path.
```

Comment:
i'll guess "unicode"? Is that the most-cross-platform way to store a thing that's a path? @vu3rdd and I were also asking ourselves "what is even the point of the mangling?" Why not just a relative path (probably "normalized to use only `/` for path-separator)?

Reply:
I guess ideally `FilePath` (or something similar) would be the most cross-platform way to store a thing that's a path :/  Both bytes and unicode have plenty of downsides and using them throughout the codebase instead of isolating them inside one abstraction leads to ... much duplication of process (to mangle a Perlis quote a bit).

I was also tempted to throw out the path mangling but I thought it would be beneficial to restrain ourselves on that to limit the scope of the changes being made all at once.  It might be good to at least delay the mangling until the network boundary in this implementation, though?  Store true paths and make `write_snapshot_to_tahoe` do the mangling (and `create_snapshot_from_capability` reverse it).

I think the *reason* the mangling exists is that Tahoe-LAFS makes naming a child including "/" in a directory hard and the old magic-folder code used the mangled name as the file name in the DMD directory.  That's *not* how we're using this path, at least not in any code I see in this PR, so maybe there's no point mangling *this* (maybe we still have to mangle it elsewhere to put it in the DMD).